In [1]:
import pandas as pd
import matplotlib.pyplot as plt

First, import the vaccines dataset. Fill in the file path to import your data.

In [ ]:
vaccines = pd.read_csv(# Fill this in )

In [22]:
vaccines = pd.read_csv('../nash-covid-cases-vaccines/data/vaccine_with_protocol.csv')

**Question:** How many rows and columns are contained in this dataset?

In [ ]:
# Your Code Here


In [3]:
vaccines.head()

vacc_date  zipcode  count_given  dose_nbr  protocol
0  2020-12-12  37215.0          1.0       1.0  two-shot
1  2020-12-14  37205.0          1.0       2.0  two-shot
2  2020-12-14  37215.0          1.0       1.0  two-shot
3  2020-12-15  37221.0          1.0       1.0  two-shot
4  2020-12-16  37215.0          1.0       1.0  two-shot

**Question:** What columns does the dataset have? What datatypes do these columns have? Are there any missing values?

In [5]:
# Your Code Here

In [4]:
vaccines.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6694 entries, 0 to 6693
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   vacc_date    6694 non-null   object 
 1   zipcode      6462 non-null   float64
 2   count_given  6686 non-null   float64
 3   dose_nbr     6686 non-null   float64
 4   protocol     6694 non-null   object 
dtypes: float64(3), object(2)
memory usage: 261.6+ KB


We don't really need the id column, as it doesn't provide any additional information. Drop this column from the dataframe.

In [6]:
# Your Code Here

**Question:** How many days does this dataset cover?

In [20]:
# Your Code Here

In [18]:
vaccines['vacc_date'].nunique()

116

In [19]:
vaccines['vacc_date'].unique()

array(['2020-12-12', '2020-12-14', '2020-12-15', '2020-12-16',
       '2020-12-17', '2020-12-18', '2020-12-19', '2020-12-20',
       '2020-12-21', '2020-12-22', '2020-12-23', '2020-12-24',
       '2020-12-25', '2020-12-26', '2020-12-27', '2020-12-28',
       '2020-12-29', '2020-12-30', '2020-12-31', '2021-01-01',
       '2021-01-02', '2021-01-03', '2021-01-04', '2021-01-05',
       '2021-01-06', '2021-01-07', '2021-01-08', '2021-01-09',
       '2021-01-10', '2021-01-11', '2021-01-12', '2021-01-13',
       '2021-01-14', '2021-01-15', '2021-01-16', '2021-01-17',
       '2021-01-18', '2021-01-19', '2021-01-20', '2021-01-21',
       '2021-01-22', '2021-01-23', '2021-01-24', '2021-01-25',
       '2021-01-26', '2021-01-27', '2021-01-28', '2021-01-29',
       '2021-01-30', '2021-01-31', '2021-02-01', '2021-02-02',
       '2021-02-03', '2021-02-04', '2021-02-05', '2021-02-06',
       '2021-02-07', '2021-02-08', '2021-02-09', '2021-02-10',
       '2021-02-11', '2021-02-12', '2021-02-13', '2021-

**Question:** How many total zip codes are included in this dataset?

In [ ]:
# Your Code Here


**Question:** Which zip code or zipcodes have the most days with at least one second-round dose? Which have the fewest?

In [ ]:
# Your Code Here


In [16]:
vaccines.loc[(vaccines.dose_nbr == 2)]['zipcode'].value_counts()

37209.0    102
37215.0    102
37205.0    101
37221.0     99
37204.0     99
37206.0     98
37212.0     98
37211.0     98
37076.0     98
37203.0     97
37208.0     96
37207.0     96
37216.0     95
37115.0     95
37072.0     95
37214.0     95
37220.0     94
37013.0     94
37217.0     93
37138.0     92
37080.0     91
37218.0     91
37210.0     91
37189.0     88
37219.0     79
37228.0     78
37201.0     72
37027.0     47
37213.0     25
37069.0     24
37238.0      2
Name: zipcode, dtype: int64

Nashville's most populous zip code is 37013. **Question:** How many rows in the dataframe correspond to this zip code 37013?

In [21]:
# Your Code Here

**Question:** What is the first day for with a positive count for dose number 2 in zipcode 37013?

In [7]:
# Your Code Here

In [14]:
vaccines.loc[(vaccines.zipcode == 37212) & 
            (vaccines.dose_nbr == 2)]

vacc_date  zipcode  count_given  dose_nbr  protocol
51    2020-12-18  37212.0          1.0       2.0  two-shot
129   2020-12-20  37212.0          1.0       2.0  two-shot
232   2020-12-23  37212.0          1.0       2.0  two-shot
417   2020-12-29  37212.0          1.0       2.0  two-shot
525   2021-01-01  37212.0          1.0       2.0  two-shot
...          ...      ...          ...       ...       ...
6355  2021-04-03  37212.0          9.0       2.0  two-shot
6424  2021-04-04  37212.0          8.0       2.0  two-shot
6500  2021-04-05  37212.0         45.0       2.0  two-shot
6583  2021-04-06  37212.0         70.0       2.0  two-shot
6664  2021-04-07  37212.0         18.0       2.0  two-shot

[98 rows x 5 columns]